openapi_python_client/resolver/resolved_schema.py

```diff
@@ -0,0 +1,202 @@
+import hashlib
+from typing import Any, Dict, Generator, List, Tuple, Union, cast
+
+from .reference import Reference
+from .resolver_types import SchemaData
+
+
+class ResolvedSchema:
+    def __init__(self, root: SchemaData, refs: Dict[str, SchemaData], errors: List[str]):
+        self._root: SchemaData = root
+        self._refs: Dict[str, SchemaData] = refs
+        self._errors: List[str] = errors
+        self._resolved_remotes_components: SchemaData = cast(SchemaData, {})
+
+        self._resolved_schema: SchemaData = cast(SchemaData, {})
+        if len(self._errors) == 0:
+            self._process()
+
+    @property
+    def schema(self) -> SchemaData:
+        return self._root
+
+    @property
+    def errors(self) -> List[str]:
+        return self._errors.copy()
+
+    def _process(self) -> None:
+        self._process_remote_paths()
+        self._process_remote_components(self._root)
+        self._root.update(self._resolved_remotes_components)
```

Comment:
The merge here is too wild, it need fixe. I was doing my test on: `https://raw.githubusercontent.com/jdegre/5GC_APIs/master/TS29504_Nudr_DataRepository.yaml`

It's `#components/securitySchemes` was erased by the ones of `_resolved_remotes_components`, leading to an invalid document. Also potentially some naming collision to handle regarding the resolved remote reference and the existing one on the `root` schema